internal/cliwrapper/cliwrapper_test.go

```diff
@@ -48,6 +49,54 @@ func TestPodman_ImageExists(t *testing.T) {
 	tests.RemoveImage(logger, tests.TestImage)
 }
 
+func TestPodman_ImageExists(t *testing.T) {
+	Podman_ImageExists(t, "")
+}
+
+func TestPodman_Remote_ImageExists(t *testing.T) {
+	var tmpPodmanSocketCmd *exec.Cmd
+
+	// Check if there is an existing connection of `podman-machine-default` since this is included when installing
+	// podman desktop for macOS.
+	connectionName := "podman-machine-default"
+	chkDefaultConnectionCmd := exec.Command(tests.GetPodmanPath(), "--connection", connectionName, "system", "info") //nolint:gosec
+	if err := chkDefaultConnectionCmd.Run(); err == nil {
+		Podman_ImageExists(t, connectionName)
+	} else if runtime.GOOS == "linux" {
+		// The podman-machine-default doesn't exist then for Linux, create a temporary socket
+
+		// Setup
+		connectionName = "arcaflow-engine-deployer-podman-test"
+		podmanSocketPath := "unix:///var/tmp/" + connectionName + ".sock"
+
+		tmpPodmanSocketCmd = exec.Command(tests.GetPodmanPath(), "system", "service", "--time=0", podmanSocketPath)
+		if err := tmpPodmanSocketCmd.Start(); err != nil {
+			t.Fatalf("Failed to create temporary podman socket")
+		}
+
+		addConnectionCmd := exec.Command(tests.GetPodmanPath(), "system", "connection", "add", connectionName, podmanSocketPath)
+		if err := addConnectionCmd.Run(); err != nil {
+			t.Fatalf("Failed to add connection: " + connectionName)
+		}
+
+		// Run test
+		Podman_ImageExists(t, connectionName)
+
+		// Clean up
+		if err := tmpPodmanSocketCmd.Process.Kill(); err != nil {
```

Comment:
You should use a `defer()` function (or [a "cleanup" function](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup)) to ensure that the service process is cleaned up.

I'll toss you a PR.

Reply:
See https://github.com/ewchong/arcaflow-engine-deployer-podman/pull/2.